File: InterviewBit/Array/MaxSumContiguousArray/MaxSumContagiousArray.cpp
int Solution::maxSubArray(const vector<int> &A) {
    int m = A.size();
    int max=INT_MIN;
    int current_sum=0;
    for (int i=0;i<m;i++){
        // cout<<sum<<endl;
        current_sum += A[i];
        if (current_sum>max) max=current_sum;
        if(current_sum<0) current_sum=0;
    }

return max;
}

File: InterviewBit/Array/MaxSumContiguousArray/MaxSumContagiousArray.py
class Solution:
    # @param A : tuple of integers
    # @return an integer
    def maxSubArray(self, A):
        max_sum = 0
        running_sum = 0
        
        for i in A:
            running_sum += i
            if running_sum > max_sum:
                max_sum = running_sum
            if running_sum<0:
                running_sum=0
        return max_sum